### PR TITLE
Warn of exceeding PyPI project size limit in GHA guide

### LINKED ISSUE #777: GitHub Actions tutorial: Warn about filling up test pypi package limit if you publish every commit as suggested
I followed the guide at https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/ and got everything working so that every push to our main branch was published to test.pypi and only actual releases were published to pypi. We were very happy for several months until the push to test.pypi started erroring because we had hit the project size limit. I went to the package page on test.pypi and discovered (unsurprisingly in retrospect) hundreds of releases published on test.pypi that accumulated to put us at our package size limit.

As far as I can tell, there does not seem to be a way either to set time limits on these releases so they are auto-deleted or to batch delete them in any way. I had to go through a very laborious process of clicking on the options button for each unwanted release, selecting `Delete` and then copying the release name into the the verification box. Please add some text to the tutorial warning users of the potential ramifications of publishing every push to test.pypi, which is what is currently suggested in the tutorial. Or if there are better ways to auto-delete or otherwise manage such test releases, please add language explaining how to do better.

### PR DESCRIPTION
**Pull request by chrysle**
Closes #777 

----
:books: Documentation preview :books:: https://python-packaging-user-guide--1380.org.readthedocs.build/en/1380/

### CODE REVIEW

**Review by willingc — APPROVED** [review]
lgtm thanks.

**Review by willingc — APPROVED** [review]
Thanks @chrysle. Merging :tada:

File: source/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows.rst [hunk 1]
```diff
@@ -219,6 +219,14 @@ And it'll publish any push to TestPyPI which is useful for
 providing test builds to your alpha users as well as making
 sure that your release pipeline remains healthy!
 
+.. attention::
+
+  If your repository has frequent commit activity and every push is uploaded
+  to TestPyPI as described, the project might exceed the
+  `PyPI project size limit <https://pypi.org/help/#project-size-limit>`_.
+  The limit could be increased, but a better solution may constitute to
+  use a PyPI-compatible server like :ref:`pypiserver` in the CI for testing purposes.
```
Comment (webknjaz) [inline source/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows.rst:228]: This suggests a different use case. The original one I had in mind was keeping the release pipeline healthy, which included going through a (Test)PyPI auth.
This isn't bad, just something to consider...

Comment (willingc) [reply source/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows.rst:228]: @webknjaz Do you want to create a PR with your intended use case? :sunny:

**Pull request merged**